spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -98,6 +98,10 @@ def reset(self):
         continue
 
       if s[0] != " " and s[0] != "\t":
+        if s.replace(" ", "").startswith("else:") or s.replace(" ", "").startswith("elif") or s.replace(" ", "").startswith("except"):
```

Comment:
My reading of the code says that it only strips off new line characters and adds them back in.  The s.replace() is never added to accumulated string -- just used for testing.

But, you may be right about not using `split("\n")` above in the code.